### PR TITLE
fix(#2098): locale short date fmt

### PR DESCRIPTION
**Pull request by vomikan**
test

---
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/2125)

### CODE REVIEW

File: src/optionsettingsgeneral.cpp [hunk 1]
```diff
@@ -175,16 +177,16 @@ void OptionSettingsGeneral::OnCurrency(wxCommandEvent& WXUNUSED(event))
...
     {
         m_date_format = data->GetData();
         m_sample_date_text->SetLabelText(wxDateTime::Now().Format(m_date_format));
     }
-    else
-    {
-        return;
-    }
+
+    return;
```
Comment (slodki) [inline src/optionsettingsgeneral.cpp]:
```suggestion
```

**Comment by GitCop** [comment]
Thanks for your pull request while there were the following issues

 * Commit: 238c9266e57a8ad8f6ef0048b9506846747aaf7a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by slodki** [comment]
My proposal: when date format is not set in settings table (NULL) then use system settings. Do not set any format by default (new MMEX installation will use system date format) and **do not store system date format in settings table** but use NULL for such user selection.

NULL date format will be dynamic - will always be updated to current system settings on each MMEX start.

**Comment by vomikan** [comment]
I am on the same point of view. But value in the setting table always defined.

**Comment by slodki** [comment]
Maybe `system` hard-coded value in db for the default locale date format?

**Comment by GitCop** [comment]
Thanks for your pull request while there were the following issues

 * Commit: 238c9266e57a8ad8f6ef0048b9506846747aaf7a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for your pull request while there were the following issues

 * Commit: 238c9266e57a8ad8f6ef0048b9506846747aaf7a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/moneymanagerex/moneymanagerex/wiki/GitCop

---
This message was auto-generated by https://gitcop.com

**Comment by vomikan** [comment]
> Maybe system hard-coded value in db for the default locale date format?

https://github.com/moneymanagerex/moneymanagerex/pull/2125/files#diff-dd0db5eb56b641d4ffa6bc51bf6afe66L48

**Pull request merged**